infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -197,9 +196,17 @@ public SnapshotReadMessage read(UUID syncRequestId) {
         log.info("Start Snapshot Sync replication for stream name={}, id={}", streamNameOfTag,
                 CorfuRuntime.getStreamID(streamNameOfTag));
 
-        // Wait for data to be available on the Snapshot Sync Queue if it is empty
+        // If data is not present in the snapshot sync queue:
+        // (1) check if data for any session was received in the last time window. If not, request for snapshot sync
```

Comment:
how are we checking this?  The opaque stream tracked in dataFound() is for the current session only

Reply:
We don't look at the opaque stream. instead, we have a static variable `lastDataReceivedEpoch`. This is updated everytime any session finds data.
And incase no data is received within the "timeout" window, only then can we say that application hasn't been providing data, and we request for data to them.

Reply:
ok, I see.  The last data received timestamp is shared across all sessions.